### ISSUE TITLE
creating mutable variable of reference to tuple containing immovable object causes `E0507`: cannot move out of shared reference and incorrect diagnostic

### ISSUE THREAD

**Issue by bend-n** [rust-lang/rust#124600]
### Code

```Rust
struct Hi; // this is immovable
let (mut _x,) = &(Hi,);
```

### Current output

```Shell
error[E0507]: cannot move out of a shared reference
 --> src/main.rs:3:21
  |
3 |     let (mut _x,) = &(Hi,);
  |          ------     ^^^^^^
  |          |
  |          data moved here
  |          move occurs because `_x` has type `Hi`, which does not implement the `Copy` trait
  |
help: consider borrowing the pattern binding
  |
3 |     let (ref mut _x,) = &(Hi,);
  |          +++
```

### Desired output

```Shell
error[E0507]: cannot move out of a shared reference
 --> src/main.rs:3:21
  |
3 |     let (mut _x,) = &(Hi,);
  |          ------     ^^^^^^
  |          |
  |          data moved here
  |          move occurs because `_x` has type `Hi`, which does not implement the `Copy` trait
  |
```

### Rust Version

```Shell
rustc 1.80.0-nightly (c987ad527 2024-05-01)
binary: rustc
commit-hash: c987ad527540e8f1565f57c31204bde33f63df76
commit-date: 2024-05-01
host: x86_64-unknown-linux-gnu
release: 1.80.0-nightly
LLVM version: 18.1.4
```

### Anything else?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9670d1424d20a93ed49b9bc818f4e0a4
